### PR TITLE
[SPARK-7106][MLlib][PySpark] Support model save/load in Python's FPGrowth

### PR DESCRIPTION
**Pull request by vectorijk**
## What changes were proposed in this pull request?

Python API supports mode save/load in FPGrowth
JIRA: [https://issues.apache.org/jira/browse/SPARK-7106](https://issues.apache.org/jira/browse/SPARK-7106)
## How was the this patch tested?

The patch is tested with Python doctest.

### CODE REVIEW

**Comment by vectorijk** [comment]
cc @mengxr @yanboliang  Could you take a look at this?

**Comment by mengxr** [comment]
@vectorijk Please check this commit https://github.com/apache/spark/commit/9ca79c1ece5ad139719e4eea9f7d1b59aed01b20 and update your PR. Thanks!

**Comment by vectorijk** [comment]
@mengxr Thanks, I didn't notice that. Addressed comment.
Also, I take a look at 9ca79c1 and it only moved cleanup temp file code for doctest under ml directory. Meanwhile, it seems like we should do the same thing under mllib. Should we also need to create a JIRA issue for this?

_off-topic_

I was wondering if Spark community would be interested in mentoring students for Google Summer of Code(GSoC) under Apache Software Foundation this year. Last time, I was impressed by the MechCoder's project mentored by @mengxr. Therefore, I look forward to having a chance to do something interesting and continue to contribute to codebase during this summer.

@mengxr, Are you still interested in mentoring a MLlib PySpark related project this summer? If so, I am very willing to brainstorm with you and others on JIRA about what could be probably worked on during this summer. And I might start to write the proposal for GSoC. If there are some pre-GSoC issues related to project, I would love to work on those.

P.S Here is the [post](http://apache-spark-developers-list.1001551.n3.nabble.com/GSoC-Interested-in-GSoC-2016-ideas-td16224.html) I published on dev mailing list.

File: python/pyspark/mllib/fpm.py [hunk 1]
```diff
@@ -40,6 +41,11 @@ class FPGrowthModel(JavaModelWrapper):
     >>> model = FPGrowth.train(rdd, 0.6, 2)
     >>> sorted(model.freqItemsets().collect())
     [FreqItemset(items=[u'a'], freq=4), FreqItemset(items=[u'c'], freq=3), ...
+    >>> model_path = temp_path + "/fpg_model"
```
Comment (yanboliang) [inline python/pyspark/mllib/fpm.py]: ```/fpm``` is enough, because we only support save/load model under the old MLlib API.

Comment (vectorijk) [inline python/pyspark/mllib/fpm.py]: ok, I have done this.

**Comment by mengxr** [comment]
LGTM. Merged into master. Thanks!

For GSoC, I created https://issues.apache.org/jira/browse/SPARK-13489 to collect some project ideas. Let's move our discussion there. If I don't have time to mentor a GSoC project, other committers might be interested. Could you prepare a draft proposal and post it on the JIRA?

**Comment by vectorijk** [comment]
@mengxr, Thanks for replying!

Definitely, I will post a rough draft proposal on JIRA later.

**Pull request closed without merging**